### DISCUSSION TITLE
Dynamic import() syntax behaving differently in Parcel2

### DISCUSSION THREAD

**Discussion by MaffooBristol** [parcel-bundler/parcel#5731] [Q&A]
I have some code which works as expected in Parcel 1:
```
export default async function foo() {
  const moment = await import('moment');
  const bar = moment().set({ month: '0' });
  // ...
```

But in Parcel 2 it throws out a `moment is not a function`, which requires me to do the following:
```
export default async function foo() {
  const moment = await import('moment');
  const bar = moment.default().set({ month: '0' });
  // ...
```

I assume this is to do with the module loading system changing in some way. Should my code be updated to always use `.default` in this situation, or is there some easier way to make it work exactly the same? ie, is it by design?

**Comment by mischnic** [comment]
The various tools don't really agree what should happen in this case (because moment is a CommonJS module, not native ESM).

https://github.com/parcel-bundler/parcel/issues/5459

I do think that `moment.default()` should be the correct thing here. But we haven't decided yet what Parcel should do
